hibernate-core/src/main/java/org/hibernate/query/criteria/HibernateCriteriaBuilder.java

```diff
@@ -892,4 +892,313 @@ <Y extends Comparable<? super Y>> JpaPredicate between(
+	 *
+	 * @param order order by clause used in within group
+	 * @param argument values to join
+	 * @param separator the separator used to join the values
+	 *
+	 * @return ordered set-aggregate expression
+	 *
+	 * @see #listagg(JpaOrder, JpaPredicate, Expression, Expression) listagg
+	 */
+	JpaExpression<String> listagg(JpaOrder order, Expression<String> argument, Expression<String> separator);
+
+	/**
+	 * Create a {@code listagg} ordered set-aggregate function expression.
+	 *
+	 * @param order order by clause used in within group
+	 * @param filter optional filter clause
+	 * @param argument values to join
+	 * @param separator the separator used to join the values
+	 *
+	 * @return ordered set-aggregate expression
+	 *
+	 * @see #functionWithinGroup
+	 */
+	JpaExpression<String> listagg(
+			JpaOrder order,
+			JpaPredicate filter,
+			Expression<String> argument,
+			Expression<String> separator);
+
+	/**
+	 * Create a {@code mode} ordered set-aggregate function expression.
+	 *
+	 * @param order order by clause used in within group
+	 * @param argument argument to the function
+	 *
+	 * @return ordered set-aggregate expression
+	 *
+	 * @see #mode(JpaOrder, JpaPredicate, Expression) mode
+	 */
+	JpaExpression<?> mode(JpaOrder order, Expression<?> argument);
+
+	/**
+	 * Create a {@code mode} ordered set-aggregate function expression.
+	 *
+	 * @param order order by clause used in within group
+	 * @param filter optional filter clause
+	 * @param argument argument to the function
+	 *
+	 * @return ordered set-aggregate expression
+	 *
+	 * @see #functionWithinGroup
+	 */
+	JpaExpression<?> mode(JpaOrder order, JpaPredicate filter, Expression<?> argument);
+
+	/**
+	 * Create a {@code percentile_cont} ordered set-aggregate function expression.
+	 *
+	 * @param order order by clause used in within group
+	 * @param argument argument to the function
+	 *
+	 * @return ordered set-aggregate expression
+	 *
+	 * @see #percentileCont(JpaOrder, JpaPredicate, Expression) percentileCont
+	 */
+	JpaExpression<Integer> percentileCont(JpaOrder order, Expression<? extends Number> argument);
+
+	/**
+	 * Create a {@code percentile_cont} ordered set-aggregate function expression.
+	 *
+	 * @param order order by clause used in within group
+	 * @param filter optional filter clause
+	 * @param argument argument to the function
+	 *
+	 * @return ordered set-aggregate expression
+	 *
+	 * @see #functionWithinGroup
+	 */
+	JpaExpression<Integer> percentileCont(JpaOrder order, JpaPredicate filter, Expression<? extends Number> argument);
+
+	/**
+	 * Create a {@code percentile_disc} ordered set-aggregate function expression.
+	 *
+	 * @param order order by clause used in within group
+	 * @param argument argument to the function
+	 *
+	 * @return ordered set-aggregate expression
+	 *
+	 * @see #percentileDisc(JpaOrder, JpaPredicate, Expression) percentileDisc
+	 */
+	JpaExpression<Integer> percentileDisc(JpaOrder order, Expression<? extends Number> argument);
```

Comment:
I think this should rather be:

```suggestion
	<T> JpaExpression<T> percentileDisc(Expression<? extends Number> argument, Expression<T> sortExpression, SortOrder sortOrder, NullPrecedence nullPrecedence);
```